tests/cases/fourslash/codeFixInferFromUsageParameterLiteral.ts

```diff
@@ -0,0 +1,10 @@
+/// <reference path='fourslash.ts' />
+
+// @noImplicitAny: true
+//// function foo([|text |]) {

```

Comment:
without this fix, what do you get?

Reply:
Later: you get `{ length: void; indexOf(arg0: string): void; charAt(arg0: number): void }`.

However, hinting that `length: number` by saying `text.length > 1` gives the correct `string`.